<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>960</width>
    <height>770</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>960</width>
    <height>770</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>960</width>
    <height>770</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="ControlsBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>181</width>
      <height>731</height>
     </rect>
    </property>
    <property name="title">
     <string>Controls</string>
    </property>
    <widget class="QPushButton" name="btnNextGeneration">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>161</width>
       <height>35</height>
      </rect>
     </property>
     <property name="text">
      <string>Next Generation</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnAction">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>161</width>
       <height>35</height>
      </rect>
     </property>
     <property name="text">
      <string>Start</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnGetRandomSpace">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>161</width>
       <height>35</height>
      </rect>
     </property>
     <property name="text">
      <string>Random space</string>
     </property>
    </widget>
    <widget class="QSlider" name="sldProbQ1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>240</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="txtProbQ1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>220</y>
       <width>141</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Probability State 1: 0.00</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="chkClosed">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>280</y>
       <width>161</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Use closed space</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="groupBox_7">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>160</y>
       <width>181</width>
       <height>51</height>
      </rect>
     </property>
     <property name="title">
      <string>Cells size (pixels)</string>
     </property>
     <widget class="QSpinBox" name="spbCellSize">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>141</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>2</number>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBoxRule">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>340</y>
       <width>181</width>
       <height>341</height>
      </rect>
     </property>
     <property name="title">
      <string>Rule</string>
     </property>
     <widget class="QScrollArea" name="scrollRules">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>171</width>
        <height>261</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents_2">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>171</width>
         <height>261</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QLineEdit" name="txtRule">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>31</y>
        <width>161</width>
        <height>31</height>
       </rect>
      </property>
      <property name="placeholderText">
       <string>Write or read a rule</string>
      </property>
     </widget>
    </widget>
    <widget class="QPushButton" name="btnSetRule">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>690</y>
       <width>161</width>
       <height>35</height>
      </rect>
     </property>
     <property name="text">
      <string>Set rule</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="chkReverseRule">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>310</y>
       <width>161</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Use reverse rule</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>20</y>
      <width>761</width>
      <height>721</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>761</width>
      <height>721</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>761</width>
      <height>721</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QScrollArea" name="scrollArea">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>751</width>
       <height>681</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="cursor" stdset="0">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOn</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOn</enum>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QAbstractScrollArea::AdjustToContents</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>728</width>
        <height>658</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QLabel" name="txtGeneration">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>690</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Generation:</string>
     </property>
    </widget>
    <widget class="QLabel" name="txtnQ0Cells">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>690</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Q0 cells: </string>
     </property>
    </widget>
    <widget class="QLabel" name="txtnQ1Cells">
     <property name="geometry">
      <rect>
       <x>390</x>
       <y>690</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Q1 cells: </string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
